src/main/scala/org/scalameter/json/scalaMeterModule.scala

```diff
@@ -0,0 +1,103 @@
+package org.scalameter
+package json
+
+import scala.language.higherKinds
+
+import com.fasterxml.jackson.core.{JsonToken, JsonParser, JsonGenerator}
+import com.fasterxml.jackson.databind._
+import com.fasterxml.jackson.databind.deser.Deserializers
+import com.fasterxml.jackson.databind.deser.std.StdDeserializer
+import com.fasterxml.jackson.databind.jsontype.{TypeSerializer, TypeDeserializer}
+import com.fasterxml.jackson.databind.Module.SetupContext
+import com.fasterxml.jackson.databind.module.SimpleModule
+import com.fasterxml.jackson.databind.ser.Serializers
+import com.fasterxml.jackson.databind.ser.std.StdSerializer
+import com.fasterxml.jackson.databind.`type`.MapLikeType
+import scala.collection.generic.ImmutableMapFactory
+import scala.collection.immutable
+import org.scalameter.picklers.Pickler
+
+
+/* Serializer for map with keys whose are subtype of PicklerBasedKey */
+class PicklerBasedMapSerializer[K <: PicklerBasedKey[_], T[_K, +V] <: immutable.Map[_K, V] with immutable.MapLike[_K, V, T[_K, V]]](clazz: Class[T[K, Any]])
```

Comment:
Since there is a type bound, is the higher kinded type really necessary here?
It seems like you could drop it, and instead just pass an ordinary `Map` supertype to `serialize`.